mteb/tasks/PairClassification/por/SickBrPC.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+
+from mteb.abstasks.AbsTaskPairClassification import AbsTaskPairClassification
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+N_SAMPLES = 1000
+
+
+class SickBrPC(AbsTaskPairClassification):
+    metadata = TaskMetadata(
+        name="SICK-BR-PC",
+        dataset={
+            "path": "eduagarcia/sick-br",
+            "revision": "0cdfb1d51ef339011c067688a3b75b82f927c097",
+        },
+        description="SICK-BR is a Portuguese inference corpus, human translated from SICK",
+        reference="https://linux.ime.usp.br/~thalen/SICK_PT.pdf",
+        type="PairClassification",
+        category="s2s",
+        eval_splits=["validation", "test"],
+        eval_langs=["por-Latn"],
+        main_score="ap",
+        date=("2018-01-01", "2018-09-01"),  # rough estimate
+        form=["written"],
+        domains=["Web"],
+        task_subtypes=[],
+        license="unknown",
+        socioeconomic_status="mixed",
+        annotations_creators="human-annotated",
+        dialect=[],
+        text_creation="human-translated and localized",
+        bibtex_citation="""
+@inproceedings{real18,
+  author="Real, Livy
+    and Rodrigues, Ana
+    and Vieira e Silva, Andressa
+    and Albiero, Beatriz
+    and Thalenberg, Bruna
+    and Guide, Bruno
+    and Silva, Cindy
+    and de Oliveira Lima, Guilherme
+    and C{\^a}mara, Igor C. S.
+    and Stanojevi{\'{c}}, Milo{\v{s}}
+    and Souza, Rodrigo
+    and de Paiva, Valeria"
+  year ="2018",
+  title="SICK-BR: A Portuguese Corpus for Inference",
+  booktitle="Computational Processing of the Portuguese Language. PROPOR 2018.",
+  doi ="10.1007/978-3-319-99722-3_31",
+  isbn="978-3-319-99722-3"
+}
+        """,
+        n_samples={"validation": N_SAMPLES, "test": N_SAMPLES},
+        avg_character_length={"validation": N_SAMPLES, "test": 54.89},
+    )
+
+    def dataset_transform(self):
```

Comment:
This is a bit too much code for my taste.
The subsampling and shuffling stuff could easily be taken care of with `self.stratified_subsampling`.
I also fail to see why you would filter out neutral labels. Is that not something we want to be able to detect?
Here's a snippet that I would prefer:
```python
def label_to_num(row: dict) -> dict:
    label = row["entailment_label"]
    if label == "NEUTRAL":
        row["labels"] = 0
    if label == "ENTAILMENT":
        row["labels"] = 1
    if label == "CONTRADICTION":
        row["labels"] = 2
    return row

def dataset_transform(self):
    self.dataset = self.stratified_subsampling(self.dataset, seed=self.seed, splits=self.metadata.eval_splits, label="label", n_samples=N_SAMPLES)
    self.dataset = self.dataset.map(label_to_num)
    self.dataset = self.dataset.rename_columns({"sentence_A": "sent1", "sentence_B": "sent2"})
```

Reply:
Thanks @x-tabdeveloping. I modeled this after the `KlueNLI` and `CTKFactsNLI` which do this very similarly. As far as I understand, the PairClassification task would prefer two classes:

https://github.com/embeddings-benchmark/mteb/blob/2562644af1c1d34edbd8aa378386ac04bec43a40/mteb/tasks/PairClassification/ces/CTKFactsNLI.py#L51-L54

https://github.com/embeddings-benchmark/mteb/blob/2562644af1c1d34edbd8aa378386ac04bec43a40/mteb/tasks/PairClassification/kor/KlueNLI.py#L47-L50

Reply:
Alrght! I will look into this, but it's good for now, I see the rationale.